youtube_dl/extractor/aznude.py

```diff
@@ -0,0 +1,117 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+
+try:
+    from urlparse import urlparse
+except ImportError:
+    from urllib.parse import urlparse
+
+import re
+
+
+class AZNudeIE(InfoExtractor):
+    IE_NAME = "aznude"
+    _VALID_URL = r'https?://(?:www\.)?aznude\.com/(?:mrskin|azncdn)/[^/?]+/[^/?]+/(?P<id>.*)\.html'
+    _TEST = {
+        'url': 'https://www.aznude.com/mrskin/marisatomei/loiteringwithintent/loiteringwithintent-mcnallytomei-hd-01-hd.html',
+        'md5': '28973bf7b818edfe55677b67bc073e40',
+        'info_dict': {
+            'id': 'loiteringwithintent-mcnallytomei-hd-01-hd',
+            'ext': 'mp4',
+            'title': 'Marisa Tomei in Loitering With Intent - 01',
+            'thumbnail': 'https://cdn1.aznude.com/marisatomei/loiteringwithintent/LoiteringWithIntent-McNallyTomei-HD-01-gigantic-4.jpg',
+            'description': 'Watch Marisa Tomei\'s Breasts scene on AZNude for free (22 seconds).',
+        },
+    }
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        numeric_id = "-".join(re.findall(r'(?P<num>(?:s\d+e\d+)|(?:\d+[xX]\d+)|(?:\d+))', video_id))
+        webpage = self._download_webpage(url, video_id)
+
+        artist = self._search_regex(r'<span><a href="/view/celeb/[^/?]/[^/?]+\.html">(?P<artist>[^<]+)</a></span>',
+                                    webpage,
+                                    url,
+                                    default=None)
+        work = self._search_regex(r'in <a href="/view/movie/[^/?]/[^/?]+\.html">(?P<work>[^<]+)</a>',
+                                  webpage,
+                                  url,
+                                  default=None)
+
+        if (artist is not None) and (work is not None):
+            title = artist + " in " + work
+        else:
+            title = self._og_search_title(webpage)
+
+        return {
+            'id': video_id,
+            'title': title + " - " + numeric_id,
```

Comment:
The title is common for all videos in the same movie/show for a particular artist. Due to a lack of numeric id of videos, I had to resort to using the last component of the url (e.g. `theblockbuster-smithdoe-hd-01-hd`). While that component is unique, it also has a lot of redundant information, resulting in filenames such as: 
`John Doe in The BlockBuster-theblockbuster-smithdoe-hd-01-hd.mp4` and 
`Jane Smith in Very Long Series-verylongseries-s13e12-smith-02.mp4`. So I created a numeric id from the last component, extracting video index as well as season+episode info, and appended it to the title, making the title unique as well as easy to understand. e.g. `John Doe in The BlockBuster - 01.mp4` and `Jane Smith in Very Long Series - s13e12-02.mp4`.